### PR TITLE
[GEOT-6600] getBounds fails for compound 3d crs in postgis jdbc store

### PR DESCRIPTION
**Pull request by NielsCharlier**
<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISCompound3DTestSetup.java [hunk 1]
```diff
@@ -132,4 +135,12 @@ protected void dropPointCompound3DTable() throws Exception {
         run("DELETE FROM  GEOMETRY_COLUMNS WHERE F_TABLE_NAME = 'pointCompound3d'");
         run("DROP TABLE \"pointCompound3d\"");
     }
+
+    @Override
+    protected void setUpDataStore(JDBCDataStore dataStore) {
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISCompound3DTestSetup.java:140]: Would be best if the tests run in both cases. Could you add a new test that enables estimated extents in a try/finally block instead?

File: modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGeneric3DOnlineTest.java [hunk 2]
```diff
@@ -354,7 +354,10 @@ public void testBounds() throws Exception {
 
         // check we got the right 2d component
         Envelope expected = new Envelope(1, 5, 0, 4);
-        assertEquals(expected, env);
+        assertEquals(expected.getMinX(), env.getMinX(), 0.05);
```
Comment (aaime) [inline modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGeneric3DOnlineTest.java:357]: This code is for all databases, where an exact match is required. In the PostGIS test you're enabling estimated extends and thus get back a lower precision bbox. See my other comment, better to make a new PostGIS specific test method instead.

**Comment by NielsCharlier** [comment]
@aaime requested changes are made

**Comment by aaime** [comment]
Looks good, merged, thanks!

**Pull request merged**